src/modm/driver/display/ssd1306_impl.hpp

```diff
@@ -80,105 +119,61 @@ modm::Ssd1306<I2cMaster, Height>::writeDisplay()
 	RF_END_RETURN(this->wasTransactionSuccessful());
 }
 
-template < class I2cMaster, uint8_t Height >
-modm::ResumableResult<bool>
+template<class I2cMaster, uint8_t Height>
+inline modm::ResumableResult<bool>
 modm::Ssd1306<I2cMaster, Height>::setOrientation(glcd::Orientation orientation)
 {
-	RF_BEGIN();
-
-	if ( RF_CALL(writeCommand((orientation == glcd::Orientation::Landscape0) ?
-			Command::SetSegmentRemap127 : Command::SetSegmentRemap0)) )
+	if (orientation == glcd::Orientation::Landscape0)
+	{
+		commandBuffer[0] = HardwareConfigCommands::SegmentRemap127;
+		commandBuffer[1] = HardwareConfigCommands::ComOutputScanDirectionDecrement;
+	} else if (orientation == glcd::Orientation::Landscape180)
 	{
-		RF_RETURN_CALL(writeCommand((orientation == glcd::Orientation::Landscape0) ?
-				Command::SetComOutputScanDirectionDecrement : Command::SetComOutputScanDirectionIncrement));
+		commandBuffer[0] = HardwareConfigCommands::SegmentRemap0;
+		commandBuffer[1] = HardwareConfigCommands::ComOutputScanDirectionIncrement;
 	}
 
-	RF_END_RETURN(false);
+	return writeCommands(2);
 }
 
-template < class I2cMaster, uint8_t Height >
+template<class I2cMaster, uint8_t Height>
 modm::ResumableResult<bool>
-modm::Ssd1306<I2cMaster, Height>::configureScroll(uint8_t origin, uint8_t size,
-		ScrollDirection direction, ScrollStep steps)
+modm::Ssd1306<I2cMaster, Height>::configureScroll(uint8_t origin, uint8_t size, ScrollDirection direction, ScrollStep steps)
 {
-	RF_BEGIN();
+	// RF_BEGIN();
 
-	if (!RF_CALL(disableScroll()))
-		RF_RETURN(false);
+	// if (!RF_CALL(disableScroll()))
+	// 	RF_RETURN(false);
 
 	{
 		uint8_t beginY = (origin > 7) ? 7 : origin;
 
 		uint8_t endY = ((origin + size) > 7) ? 7 : (origin + size);
 		if (endY < beginY) endY = beginY;
 
-		commandBuffer[1] = static_cast<uint8_t>(direction);
-		commandBuffer[3] = 0x00;
-		commandBuffer[5] = beginY;
-		commandBuffer[7] = i(steps);
-		commandBuffer[9] = endY;
-		commandBuffer[11] = 0x00;
-		commandBuffer[13] = 0xFF;
+		commandBuffer[0] = uint8_t(direction);
+		commandBuffer[1] = 0x00;
+		commandBuffer[2] = beginY;
+		commandBuffer[3] = uint8_t(steps);
+		commandBuffer[4] = endY;
+		commandBuffer[5] = 0x00;
+		commandBuffer[6] = 0xFF;
 	}
 
-	RF_WAIT_UNTIL( startTransactionWithLength(14) );
-
-	RF_WAIT_WHILE(this->isTransactionRunning());
-
-	RF_END_RETURN(this->wasTransactionSuccessful());
+	return writeCommands(7);
```

Comment:
Without RF_BEGIN/END, the whole function is executed again for every polling call!